packages/compass-schema/src/modules/schema-analysis.ts

```diff
@@ -72,7 +72,10 @@ export const analyzeSchema = async (
       : {
           signal: abortSignal,
         };
-    const schemaAccessor = await analyzeDocuments(docs, schemaParseOptions);
+    const schemaAccessor = await analyzeDocuments(
+      sampleCursor,
```

Comment:
This could impact performance. I have not tested that.

Reply:
Any specific concerns or predictions for where the biggest impact might be? I def defer to another team member to give you an answer here, but as a newbie what would it entail to validate at least the biggest bottlenecks we think may come up here?

Reply:
Sorry didn't get back sooner. I agree, we've shifted the cost but in a way that should leave the app a bit more responsive as it acts on a document at a time. Plus this takes advantage of the driver's lazy deserialize. In the end the time spent should be the equivalent, broken up over more event loop cycles